pkg/providers/tinkerbell/reconciler/reconciler.go

```diff
@@ -97,6 +99,44 @@ func (r *Reconciler) ReconcileWorkerNodes(ctx context.Context, log logr.Logger,
 	return controller.Result{}, nil
 }
 
+// ValidateDatacenterConfig updates the cluster status if the TinkerbellDatacenter status indicates that the spec is invalid.
+func (r *Reconciler) ValidateDatacenterConfig(ctx context.Context, log logr.Logger, clusterSpec *c.Spec) (controller.Result, error) {
+	log = log.WithValues("phase", "validateDatacenterConfig")
+	dataCenterConfig := clusterSpec.TinkerbellDatacenter
+
+	if clusterSpec.Cluster.IsManaged() {
+
+		// for workload cluster tinkerbell IP must match management cluster tinkerbell IP
+		managementClusterSpec := &anywherev1.Cluster{}
+
+		err := r.client.Get(ctx, client.ObjectKey{
+			Namespace: clusterSpec.Cluster.Namespace,
+			Name:      clusterSpec.Cluster.Spec.ManagementCluster.Name,
+		}, managementClusterSpec)
+		if err != nil {
+			return controller.Result{}, err
+		}
+
+		managementDatacenterConfig := &anywherev1.TinkerbellDatacenterConfig{}
+
+		err = r.client.Get(ctx, client.ObjectKey{
+			Namespace: clusterSpec.Cluster.Namespace,
+			Name:      managementClusterSpec.Spec.DatacenterRef.Name,
+		}, managementDatacenterConfig)
+		if err != nil {
+			return controller.Result{}, err
+		}
+
+		if dataCenterConfig.Spec.TinkerbellIP != managementDatacenterConfig.Spec.TinkerbellIP {
+			log.Error(err, "Invalid TinkerbellDatacenterConfig")
```

Comment:
err should be nil here.

Reply:
You're right, updating this log statement

Reply:
I ended up splitting the validation logic into it's own function used this log statement to catch the error from that.